src/main.cc

```diff
@@ -296,8 +318,653 @@ uint64_t rotr64 (uint64_t n, unsigned int c)
     return (n>>c) | (n<<( (-c)&mask ));
 }
 
+/*
+Georgios's Page Walker
+Author: Georgios Vavouliotis
+Advisors: Lluc Alvarez, Marc Casas Guix
+{georgios.vavouliotis, lluc.alvarez, marc.casas}@bsc.es
+
+Ported by James Coman - james in the domain tamu.edu
+*/
+int simulated_page_walker(uint32_t cpu, uint64_t vpage, int swap, uint8_t is_code, uint64_t instr_id, uint64_t ip, int type){
+
+	// 4KB: LOG2_PAGE_SIZE == 12
+	// 2MB: LOG2_PAGE_SIZE == 21
+
+	int cstall = 2; // the latency cost of the current page walk
```

Comment:
Here, why not using the same types as in the method parameters? That would add consistency to the code.

```suggestion
int32_t simulated_page_walker(uint32_t cpu, uint64_t vpage, int swap, uint8_t is_code, uint64_t instr_id, uint64_t ip, int type){

	// 4KB: LOG2_PAGE_SIZE == 12
	// 2MB: LOG2_PAGE_SIZE == 21

	int32_t cstall = 2; // the latency cost of the current page walk
```

Reply:
I think an argument can be made here that the width of the integer does not matter because this variable is not microarchitecture. So, we can simply let the compiler choose the width. I think we generally underuse `int`, `bool`, etc in ChampSim in situations where it would communicate better.